Wrappers/Python/ccpi/astra/processors/AstraBackProjector3D.py

```diff
@@ -0,0 +1,67 @@
+from ccpi.framework import DataProcessor, ImageData, AcquisitionData
+from ccpi.astra.utils import convert_geometry_to_astra
+import astra
+
+class AstraBackProjector3D(DataProcessor):
+    '''AstraBackProjector3D
+    
+    Back project AcquisitionData to ImageData using ASTRA proj_geom, vol_geom.
+    
+    Input: AcquisitionData
+    Parameter: proj_geom, vol_geom
+    Output: ImageData
+    '''
+    
+    def __init__(self,
+                 volume_geometry=None,
+                 sinogram_geometry=None,
+                 proj_geom=None,
+                 vol_geom=None,
+                 output_axes_order=None):
+        kwargs = {
+                  'volume_geometry'  : volume_geometry,
+                  'sinogram_geometry'  : sinogram_geometry,
+                  'proj_geom'  : proj_geom,
+                  'vol_geom'  : vol_geom,
+                  'output_axes_order'  : output_axes_order
+                  }
+        
+        #DataProcessor.__init__(self, **kwargs)
+        super(AstraBackProjector3D, self).__init__(**kwargs)
+        
+        self.set_ImageGeometry(volume_geometry)
+        self.set_AcquisitionGeometry(sinogram_geometry)
+        
+                # Set up ASTRA Volume and projection geometry, not to be stored in self
+        vol_geom, proj_geom = convert_geometry_to_astra(self.volume_geometry,
+                                                        self.sinogram_geometry)
+        
+        # Also store ASTRA geometries
+        self.vol_geom = vol_geom
+        self.proj_geom = proj_geom
+    
+    def check_input(self, dataset):
+        if dataset.number_of_dimensions == 3:
+               return True
+        else:
+            raise ValueError("Expected input dimensions is 3, got {0}"\
+                             .format(dataset.number_of_dimensions))
+        
+    def set_ImageGeometry(self, volume_geometry):
+        self.volume_geometry = volume_geometry
+        
+    def set_AcquisitionGeometry(self, sinogram_geometry):
+        self.sinogram_geometry = sinogram_geometry
+    
+    def process(self):
```

Comment:
could you add the `out` parameter? see [here](https://github.com/vais-ral/CCPi-Framework/blob/composite_operator_datacontainer/Wrappers/Python/ccpi/framework/framework.py#L1123) and [here](https://github.com/vais-ral/CCPi-Framework/blob/composite_operator_datacontainer/Wrappers/Python/ccpi/framework/framework.py#L1077)

I'd say something minimal like this should do

```python

def process(self, out=None):
    ... the processing returns some ret
    if out not is None:
        out.fill(ret)
```

Reply:
or even `raise` an `NotImplementedError()` could work.